test/unit/dialog-service.spec.js

```diff
@@ -227,4 +227,49 @@ describe('the Dialog Service', () => {
         done();
       });
   });
+
+  it('".open" properly propagates errors', (done) => {
+    let catchWasCalled = false;
+    let promise = dialogService.open({ viewModel: 'test/fixtures/non-existent' })
+      .catch(() => {
+        catchWasCalled = true;
+      }).then(() => {
+        expect(catchWasCalled).toBe(true);
+        done();
+      });
+  });
+
+  it('reports no active dialog after ".closeAll" has been invoked', (done) => {
+    const settings = { viewModel: TestElement };
+    const dialogsToOpen = 6;
+    let i = dialogsToOpen;
+    let controllersPromises = [];
+    let controllers;
+    let catchWasCalled = false;
+
+    expect(dialogService.hasActiveDialog).toBe(false);
```

Comment:
Semantics here but we can move this to a separate describe or if block to state that this property resolves to true regardless.  This tests the unit more than the integration as well.  Not a stickler here though so perhaps we can merge this first and I can help to refactor the test in a new PR to demonstrate.